lib/ansible/modules/system/aix_lvol.py

```diff
@@ -0,0 +1,354 @@
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+ANSIBLE_METADATA = {'status': ['preview'],
+                    'supported_by': 'community',
+                    'version': '1.0'}
+
+DOCUMENTATION = '''
+---
+author:
+    - "Alain Dejoux (@adejoux)"
+module: aix_lvol
+short_description: Configure AIX LVM logical volumes
+description:
+  - This module creates, removes or resizes AIX logical volumes. Inspired by lvol module.
+version_added: "2.3"
+options:
+  vg:
+    description:
+    - The volume group this logical volume is part of.
+    required: true
+  lv:
+    description:
+    - The name of the logical volume.
+    required: true
+  lv_type:
+    description:
+    - The type of the logical volume. Default to jfs2.
+    required: false
+  size:
+    description:
+    - The size of the logical volume, size by default in megabytes or optionally with one of the [MGT] units.
+  copies:
+    description:
+    - the number of copies of the logical volume. By default, 1 copy. Maximum copies are 3.
+    required: false
+  policy:
+    choices: [ "maximum", "minimum" ]
+    default: maximum
+    description:
+    - Sets the interphysical volume allocation policy. "maximum" allocates logical partitions across the maximum number of physical volumes.
+      "minimum" allocates logical partitions across the minimum number of physical volumes.
+    required: false
+  state:
+    choices: [ "present", "absent" ]
+    default: present
+    description:
+    - Control if the logical volume exists. If C(present) and the
+      volume does not already exist then the C(size) option is required.
+    required: false
+  opts:
+    description:
+    - Free-form options to be passed to the mklv command
+  pvs:
+    description:
+    - Comma separated list of physical volumes e.g. hdisk1,hdisk2
+    required: false
+'''
+
+EXAMPLES = '''
+# Create a logical volume of 512M.
+- aix_lvol:
+    vg: testvg
+    lv: testlv
+    size: 512M
+
+# Create a logical volume of 512M with disks hdisk1 and hdisk2
+- aix_lvol:
+    vg: testvg
+    lv: test2lv
+    size: 512M
+    pvs: hdisk1,hdisk2
+
+# Create a logical volume of 512M mirrored.
+- aix_lvol:
+    vg: testvg
+    lv: test3lv
+    size: 512M
+    copies: 2
+
+# Create a logical volume of 1G with a minimum placement policy .
+- aix_lvol:
+    vg: rootvg
+    lv: test4lv
+    size: 1G
+    policy: minimum
+
+# Create a logical volume with special options like mirror pool
+- aix_lvol:
+    vg: testvg
+    lv: testlv
+    size: 512M
+    opts: -p copy1=poolA -p copy2=poolB
+
+# Extend the logical volume to 1200M.
+- aix_lvol:
+    vg: testvg
+    lv: test4lv
+    size: 1200M
+
+
+# Remove the logical volume.
+- aix_lvol:
+    vg: testvg
+    lv: testlv
+    state: absent
+'''
+
+RETURN = '''
+msg:
+  type: string
+  description: A friendly message describing the task result.
+  returned: always
+  sample: Logical volume testlv created.
+'''
+
+import re
+
+def convert_size(size):
```

Comment:
Like it :) I will implement something like that along your others recommendations. thanks